src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can use the destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ..... } = action;
  
    switch (type) {
```